articles/112_generated_interfaces_cpp.md

```diff
@@ -47,9 +47,43 @@ For each message two files are being generated:
 - `<my_message_name>.hpp` currently only includes `<my_message_name>__struct.hpp`
 - `<my_message_name>__struct.hpp` containing the definition of the struct
 
-This allow to add addition files beside the one with the suffix `__struct` to provide additional functionality.
+This allows to add additional files besides the one with the suffix `__struct` to provide additional functionality.
 For each additional functionality it can be decided to include it from the first header file.
 
+<div class="alert alert-warning" markdown="1">
+  <b>TODO:</b> specify content of <code>&lt;my_message_name&gt;__traits.hpp</code> file
+</div>
+
+### Types
+
+#### Mapping of primitive types
+
+| ROS type | C++ type    |
+| -------- | ----------- |
+| bool     | bool        |
+| byte     | uint8_t     |
+| char     | char        |
+| float32  | float       |
+| float64  | double      |
+| int8     | int8_t      |
+| uint8    | uint8_t     |
+| int16    | int16       |
+| uint16   | uint16      |
+| int32    | int32       |
+| uint32   | uint32      |
+| int64    | int64       |
+| uint64   | uint64_t    |
+| string   | std::string |
+
+#### Mapping of arrays and bounded strings
+
+| ROS type                | C++ type         |
+| ----------------------- | ---------------- |
+| static array            | std::array<T, N> |
+| unbounded dynamic array | std::vector<T>   |
+| bounded dynamic array   | std::vector<T>   |
```

Comment:
Indeed, but I think we have to revisit this. An array always has N elements but we need to be able to store explicitly only M (<N) elements (and are able to get the count M)..